### ISSUE TITLE
Allow pub visibility for Parser 

### ISSUE THREAD

**Issue by Emilgardis** [Nadrieril/pest_consume#2]
Currently, if you want to do

```rust
#[derive(Parser)]
#[grammar = "grammar.pest"]
pub struct MyParser;

#[pest_consume::parser]
impl MyParser { ... }
```

You get the following error:

```text
5 | #[pest_consume::parser]
   | -^^^^^^^^^^^^^^^^^^^^^^
   | |
   | can't leak private type
   | `parser::AliasedRule` declared as private
```

Is this expected or am I doing something wrong here? Visibility shouldn't matter imo.

**Comment by Nadrieril** [comment]
Yep you're right that's a bug

**Comment by Nadrieril** [comment]
Fix is available in version `1.0.4` :)
